### DISCUSSION TITLE
How do I do a search and get useful information back out of the RPC API?

### DISCUSSION THREAD

**Discussion by oliwarner** [LMS-Community/slimserver#933] [Q&A] [answered]
I _can_ `["search" .. .. "extended:1"]` and that quickly returns album/track/contributor names, IDs and coverid, but it doesn't appear to be configurable. At a bare minimum, for albums and tracks, I need artist names. Tracks need album names too.

I can hit each one for more information. But now I'm smashing the API with n+1 queries each search.

Is there any way to do this sort of search in in one query through the API?

I note that the frontend doesn't use the RPC, it generates its own HTML server-side.

**Comment by mherger** [comment, marked as answer]
Instead of running an additional request per search result, you could run individual searches for those categories: instead of running `search`, then one `albums` per album, you do a `albums 0 10 search:xyz`, `artists 0 10 search:xyz`, `tracks 0 10 search:xyz`. It's three searches, a bit slower, but would give you the additional information you want quicker than going through the results of `search` and run individual lookups. `search` really was optimized for speed, to get results while you type. Therefore it tries to avoid looking up data in other tables.

**Reply by oliwarner** [reply]
So replace my `search` command with three detailed commands. Makes sense.

I've just tried that and this is my ignorance showing but how do I get it to return more tagged parameters? I can see `albums` supports a big pile of them but I'm not sure how to specify them in my command. I'm only getting ID and album (the title) by default.

**Reply by mherger** [reply]
Use the `tags:...` parameter. Eg.

```
albums 0 1 tags:tay
albums 0 1 tags:tay id:112808 year:2013 title:The 20/20 Experience artist:Justin Timberlake count:121
```

See http://htmlpreview.github.io/?https://raw.githubusercontent.com/Logitech/slimserver/public/8.4/HTML/EN/html/docs/cli-api.html#albums. This documentation is shipped with your LMS: http://localhost:9000/html/docs/cli-api.html (or whatever host name).

**Reply by oliwarner** [reply]
Thank you, thank you, thank you. I have been reading the CLI documentation but it's pretty dense stuff. None of the commands I was looking at used the `tags:` argument and I couldn't see it in the introduction. I'm just making excuses for myself. Thank you again, for this and everything else.
